lib/myob/api/client.rb

```diff
@@ -1,69 +1,43 @@
-require 'base64'
-require 'oauth2'
-
 module Myob
   module Api
     class Client
-      include Myob::Api::Helpers
-
-      attr_reader :current_company_file, :client
 
-      def initialize(options)
-        Myob::Api::Model::Base.subclasses.each {|c| model(c.name.split("::").last)}
+      extend Myob::Api::Helpers
 
-        @redirect_uri         = options[:redirect_uri]
-        @consumer             = options[:consumer]
-        @access_token         = options[:access_token]
-        @refresh_token        = options[:refresh_token]
-        @current_company_file = options[:selected_company_file] || {}
-        @client               = OAuth2::Client.new(@consumer[:key], @consumer[:secret], {
-          :site          => 'https://secure.myob.com',
-          :authorize_url => '/oauth2/account/authorize',
-          :token_url     => '/oauth2/v1/authorize',
-        })
-
-        if options[:company_file]
-          @current_company_file = select_company_file(options[:company_file])
+      class << self
+        def define_model_method(model_class)
+          model_name = model_class.name.split('::').last
+          define_method(underscore(model_name)) do
+            Myob::Api::Query.new(model_class)
+          end
         end
-      end
 
-      def get_access_code_url(params = {})
-        @client.auth_code.authorize_url(params.merge(scope: 'CompanyFile', redirect_uri: @redirect_uri))
+        def models
+          @models ||= [
+            Myob::Api::Model::Contact,
```

Comment:
Yup.

Not sure if it's better to make it explicit about which objects are loaded, or should make use of ObjectSpace like @BrunoChauvet does here https://github.com/davidlumley/myob-api/pull/19/files#diff-0107adb2ba05918844bfaaba5ec45f18R12